### PR TITLE
first pass at autopilot cmd

### PR DESCRIPTION
**Pull request by lanzafame**

### CODE REVIEW

File: cmd/agent_autopilot.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+				chainHeadHeight, err := PoolsSDK.Query().ChainHeight(cmd.Context())
+				if err != nil {
+					log.Println(err)
+				}
+
+				// calculate epoch frequency
+				// frequency is in days so multiply by 24 hours, 60 minutes, 60 seconds to get seconds
+				// divide by 30 seconds to get epochs
+				epochFreq := big.NewFloat(float64(frequency * 24 * 60 * 60 / 30))
```
Comment (Schwartz10) [inline cmd/agent_autopilot.go:66]: Should be able to use https://github.com/glifio/go-pools/blob/bcdd3dac557e3e049979532ee55c19d643b34f84/constants/constants.go#L16 here

**Review by Schwartz10 — CHANGES_REQUESTED** [review]
Overall looks good, small changes requested

File: cmd/agent_autopilot.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+				// each loop retrieve config values aka hot-reload
+				paymentType := viper.GetString("autopilot.payment-type")
+				log.Println("Payment type: ", paymentType)
+				if paymentType == "principal" || paymentType == "custom" {
+					amount := viper.GetInt64("autopilot.amount")
+					args = append(args, fmt.Sprintf("%d", amount))
+				}
+
+				//TODO: maybe change frequency to max debt or max epoch difference
+				frequency := viper.GetFloat64("autopilot.frequency")
```
[the remark this answers is not in the document]
Comment (lanzafame) [reply cmd/agent_autopilot.go:46]: Correct.

File: cmd/agent_autopilot.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+				log.Println("Shutting down...")
+				Exit(0)
+			default:
+				log.Println("Checking for payments...")
+				// each loop retrieve config values aka hot-reload
+				paymentType := viper.GetString("autopilot.payment-type")
+				log.Println("Payment type: ", paymentType)
+				if paymentType == "principal" || paymentType == "custom" {
+					amount := viper.GetInt64("autopilot.amount")
+					args = append(args, fmt.Sprintf("%d", amount))
```
[the remark this answers is not in the document]
Comment (lanzafame) [reply cmd/agent_autopilot.go:42]: Yes, I will fix that.

File: cmd/agent_autopilot.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+			select {
+			case <-sigs:
+				log.Println("Shutting down...")
+				Exit(0)
+			default:
+				log.Println("Checking for payments...")
+				// each loop retrieve config values aka hot-reload
+				paymentType := viper.GetString("autopilot.payment-type")
+				log.Println("Payment type: ", paymentType)
+				if paymentType == "principal" || paymentType == "custom" {
```
[the remark this answers is not in the document]
Comment (lanzafame) [reply cmd/agent_autopilot.go:40]: I just hit this issue. So yes I will implement this now.

File: config.toml [hunk 5]
```diff
@@ -1,3 +1,9 @@
 [daemon]
 rpc-url = 'https://api.node.glif.io/rpc/v1'
 token = ''
+
+[autopilot]
+payment-type = 'to-current'
```
Comment (lanzafame) [inline config.toml:6]:
```suggestion
# <to-current|principal|custom>
payment-type = 'to-current'
```

**Pull request merged**